lib/Doctrine/ORM/Internal/Hydration/AbstractHydrator.php

```diff
@@ -151,9 +152,8 @@ public function iterate($stmt, $resultSetMapping, array $hints = [])
     /**
      * Initiates a row-by-row hydration.
      *
-     * @return iterable<mixed>
-     *
      * @psalm-param array<string, mixed> $hints
+     * @psalm-return Generator<int, (array|mixed), mixed, void>
```

Comment:
What does this syntax mean? I'd expect no more than one comma inside the first level of `<>`

Reply:
Generator is kinda special and Psalm support up to 4 templated types for it: https://psalm.dev/docs/annotating_code/type_syntax/object_types/
Strangely, Psalm's doc says that the last two are mixed by default, but Psalter generated a void in 4th position instead of mixed. It doesn't seem to provoke errors up to Psalm lvl 4 one way or another, I removed the last two parameters

Reply:
Oh right, how did I forget about this 👍 Anyway here it doesn't matter indeed, removing them is good.